pkg/webhook/image_swapper.go

```diff
@@ -152,11 +152,11 @@ func (p *ImageSwapper) Mutate(ctx context.Context, obj metav1.Object) (bool, err
 		switch p.imageSwapPolicy {
 		case types.ImageSwapPolicyAlways:
 			log.Ctx(lctx).Debug().Str("image", targetImage).Msg("set new container image")
-			pod.Spec.Containers[i].Image = targetImage
+			container.Image = targetImage
```

Comment:
I don't believe this or line 159 will function as expected. Unless I am mistaken, this function's purpose is to actually alter `obj`. Since `pod.Spec.Containers` is a `[]Container` rather than `[]*Container`, this code will be setting `.Image` on a copy of the container and the original `obj`'s copy of the container will be unaltered. This is why the code before the change was indexing into the slice and setting the value on the original container instance. Here's an [example](https://play.golang.org/p/CJV02_uTKFR) of the problem if my attempt at an explanation didn't make sense.

One solution would be to extract the loop out to a function and call it twice (once for `pod.Spec.Containers` and once for `pod.Spec.InitContainers`).

Reply:
Ah - I had a feeling that was why it was the way before

Wanted to avoid refactoring out too much

Reply:
Will update and push

Reply:
@estahn pushed

Reply:
lmk how it looks now

Reply:
These lines are always updating `pod.Spec.Containers[i]` regardless of what was passed in. It should be `containers[i]`. As it currently stands, it will change the container images when passing in init containers (or crash if there are more init containers than containers).

https://github.com/estahn/k8s-image-swapper/blob/d6e70ceb5b708e807a4c726679900619509bd0ef/pkg/webhook/image_swapper.go#L160
https://github.com/estahn/k8s-image-swapper/blob/d6e70ceb5b708e807a4c726679900619509bd0ef/pkg/webhook/image_swapper.go#L164

Reply:
ah, yes

pushed